baseplate/lib/events.py

```diff
@@ -93,10 +94,15 @@ class EventQueue(ContextFactory, config.Parser, Generic[T]):
 
     """
 
-    def __init__(self, name: str, event_serializer: Callable[[T], bytes]):
-        self.queue = MessageQueue(
-            "/events-" + name, max_messages=MAX_QUEUE_SIZE, max_message_size=MAX_EVENT_SIZE
-        )
+    def __init__(
+        self, name: str, event_serializer: Callable[[T], bytes], queue_type: str = "posix"
+    ):
+        if queue_type == "in_memory":
```

Comment:
I still think it would be better to have a `QueueType` [enum](https://docs.python.org/3/library/enum.html?highlight=enum#module-enum) class for this instead of having to remember the correct text string every time.
```python
from enum import Enum

class QueueType(Enum):
    IN_MEMORY = "in_memory"
    REMOTE = "remote"
    POSIX = "posix"
```
Then you could do `if queue_type == QueueType.IN_MEMORY.value:` or better yet just accept a QueueType enum as the argument `queue_type: QueueType = QueueType.POSIX`.

Reply:
great suggestion!! done, thanks